dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -0,0 +1,272 @@
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+constexpr static const char* kSigmaArgName = "sigma";
+constexpr static const char* kWindowSizeArgName = "window_size";
+
+DALI_SCHEMA(GaussianBlur)
+    .DocStr(R"code(Apply Gaussian Blur to the input.
+Separable convolution with Gaussian Kernel is used to calculate the output.
+
+User can specify the sigma or kernel window size.
+If only the sigma is provided, the radius is of kernel is calculated ``ceil(3 * sigma)``,
+thus the kernel window size is ``2 * ceil(3 * sigma) + 1``.
+
+If only the kernel window size is provided, the sigma is calculated using the following formula::
+
+  radius = (window_size - 1) / 2
+  sigma = (radius - 1) * 0.3 + 0.8
+
+Both sigma and kernel window size can be specified as single value for all data axes
+or per data axis.
+
+When specifying the sigma or window size per axis, they are provided same as layouts: from outermost
+to innermost.
+The channel ``C`` and frame ``F`` dimensions are not considered data axes.
+
+For example, with ``HWC`` input, user can provide ``sigma=1.0`` or ``sigma=(1.0, 2.0)`` as there
+are two data axes H and W.
+
+The same input can be provided as per-sample tensors.
+)code")
+    .NumInput(1)
+    .NumOutput(1)
+    .AllowSequences()
+    .SupportVolumetric()
+    .AddOptionalArg(kWindowSizeArgName, "The diameter of kernel.", std::vector<int>{0}, true)
+    .AddOptionalArg<float>(kSigmaArgName, R"code(Sigma value for Gaussian Kernel.)code",
+                           std::vector<float>{0.f}, true);
+
+/**
+ * @brief Fill the result span with the argument which can be provided as:
+ * * ArgumentInput - {result.size()}-shaped Tensor
+ * * ArgumentInput - {1}-shaped Tensor, the value will be replicated `result.size()` times
+ * * Vector input - single "repeated argument" of length {result.size()} or {1}
+ * * scalar argument - it will be replicated `result.size()` times
+ *
+ * TODO(klecki): we may want to make this a generic utility and propagate the span-approach to
+ * the rest of the related argument gettters
+ */
+template <typename T>
+void GetGeneralizedArg(span<T> result, const std::string name, int sample_idx, const OpSpec& spec,
+                       const ArgumentWorkspace& ws) {
+  int argument_length = result.size();
+  if (spec.HasTensorArgument(name)) {
+    const auto& tv = ws.ArgumentInput(name);
+    const auto& tensor = tv[sample_idx];
+    DALI_ENFORCE(tensor.shape().sample_dim() == 1,
+                 make_string("Argument ", name, " for sample ", sample_idx,
+                             " is expected to be 1D, got: ", tensor.shape().sample_dim(), "."));
+    DALI_ENFORCE(tensor.shape()[0] == 1 || tensor.shape()[0] == argument_length,
+                 make_string("Argument ", name, " for sample ", sample_idx,
+                             " is expected to have shape equal {1} or {", argument_length,
+                             "}, got: ", tensor.shape(), "."));
```

Comment:
```suggestion
                             "} as the number of data input dimensions, got: ", tensor.shape(), "."));
```

Reply:
It's not the dimension, the dimension is (rank, or whatever) is 1 (1D). 
We check the actual shape here, we expect either 1 number, or `argument_length` numbers (as we can get either uniform scalar argument or a "vector"/repeated argument for every sample).

Reply:
So please reword.

Reply:
I think it provides the shapes as it should, unless I don't see somethings.